homeassistant/components/fronius/__init__.py

```diff
@@ -127,19 +155,16 @@ async def _get_inverter_infos(self) -> list[FroniusDeviceInfo]:
                     unique_id=unique_id,
                 )
             )
-        if not self.solar_net_device_id:
-            first_inverter_uid: str = inverter_infos[0].unique_id
-            self.solar_net_device_id = f"solar_net_{first_inverter_uid}"
-
         return inverter_infos
 
     @staticmethod
     async def _init_optional_coordinator(
         coordinator: FroniusCoordinatorType,
     ) -> FroniusCoordinatorType | None:
         """Initialize an update coordinator and return it if devices are found."""
-        await coordinator.async_refresh()
-        if coordinator.last_update_success is False:
+        try:
+            await coordinator.async_config_entry_first_refresh()
+        except ConfigEntryNotReady:
```

Comment:
Is this needed? I'm not sure if we should catch `ConfigEntryNotReady` in the integration code for other reasons than to make sure resources are properly handled (closed). And even then this should get reraised

Reply:
It's catched because it is an optional coordinator - we just try to query the endpoint and if it fails there is we know there is no such device (e.g. battery). If it succeeds we keep the coordinator. 

Since `async_config_entry_first_refresh` catches `Exception` and raises `ConfigEntryNotReady` from it we can't just catch `FroniusError` here. 

The upside of this approach is that we only need 1 query to determine if a device is present and if it is we already have a data-initialized coordinator ready to construct entities from. 

If the bridge itself is down `ConfigEntryNotReady` is already raised at this point from the mandatory coordinators (logger and inverter).